builder/src/build/patches/investigated/update-webpack-chunks-file/get-file-content-with-updated-webpack-require-f-code.ts

```diff
@@ -0,0 +1,74 @@
+import * as ts from "ts-morph";
+
+export async function getFileContentWithUpdatedWebpackFRequireCode(
+  sourceFile: ts.SourceFile,
+  { installChunk, installedChunks }: { installChunk: string; installedChunks: string },
+  chunks: string[]
+): Promise<string> {
+  const webpackFRequireFunction = sourceFile
+    .getDescendantsOfKind(ts.SyntaxKind.BinaryExpression)
+    .map((binaryExpression) => {
+      const binaryExpressionLeft = binaryExpression.getLeft();
+      if (!binaryExpressionLeft.getText().endsWith(".f.require")) return;
+
+      const binaryExpressionOperator = binaryExpression.getOperatorToken();
+      if (binaryExpressionOperator.getText() !== "=") return;
+
+      const binaryExpressionRight = binaryExpression.getRight();
+      const binaryExpressionRightText = binaryExpressionRight.getText();
+
+      if (
+        !binaryExpressionRightText.includes(installChunk) ||
+        !binaryExpressionRightText.includes(installedChunks)
+      )
+        return;
+
+      if (!binaryExpressionRight.isKind(ts.SyntaxKind.ArrowFunction)) return;
+
+      const arrowFunctionBody = binaryExpressionRight.getBody();
+      if (!arrowFunctionBody.isKind(ts.SyntaxKind.Block)) return;
+
+      const arrowFunction = binaryExpressionRight;
+      const functionParameters = arrowFunction.getParameters();
+      if (functionParameters.length !== 2) return;
+
+      const callsInstallChunk = arrowFunctionBody
+        .getDescendantsOfKind(ts.SyntaxKind.CallExpression)
+        .some((callExpression) => callExpression.getExpression().getText() === installChunk);
+      if (!callsInstallChunk) return;
+
+      const functionFirstParameterName = functionParameters[0]?.getName();
+      const accessesInstalledChunksUsingItsFirstParameter = arrowFunctionBody
+        .getDescendantsOfKind(ts.SyntaxKind.ElementAccessExpression)
+        .some((elementAccess) => {
+          return (
+            elementAccess.getExpression().getText() === installedChunks &&
+            elementAccess.getArgumentExpression()?.getText() === functionFirstParameterName
+          );
+        });
+      if (!accessesInstalledChunksUsingItsFirstParameter) return;
+
+      return arrowFunction;
+    })
+    .find(Boolean);
```

Comment:
Just for my own clarity this is effectively returning the first item in the array that is not falsy?

Reply:
Yes, if this code looks unclear I can try to improve it

Reply:
I've refactored the code to remove the `find(Boolean)` 🙂👍